packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/drop/useDrop.tsx

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { gql, useMutation } from '@apollo/client';
+import { DRAG_SOURCES_TYPE, useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import { useContext, useEffect } from 'react';
+import { DiagramContext } from '../../contexts/DiagramContext';
+import { DiagramContextValue } from '../../contexts/DiagramContext.types';
+import {
+  GQLDropOnDiagramData,
+  GQLDropOnDiagramPayload,
+  GQLDropOnDiagramVariables,
+  GQLErrorPayload,
+  UseDropValue,
+} from './useDrop.types';
+
+const dropOnDiagramMutation = gql`
+  mutation dropOnDiagram($input: DropOnDiagramInput!) {
+    dropOnDiagram(input: $input) {
+      __typename
+      ... on DropOnDiagramSuccessPayload {
+        diagram {
+          id
+        }
+      }
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload: GQLDropOnDiagramPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+export const useDrop = (): UseDropValue => {
+  const { addErrorMessage } = useMultiToast();
+  const { diagramId, editingContextId } = useContext<DiagramContextValue>(DiagramContext);
+  const [dropMutation, { data: droponDiagramElementData, error: droponDiagramError }] = useMutation<
+    GQLDropOnDiagramData,
+    GQLDropOnDiagramVariables
+  >(dropOnDiagramMutation);
+
+  const onDrop = (event: React.DragEvent, diagramElementId?: string) => {
+    event.preventDefault();
+    const type = event.dataTransfer.getData(DRAG_SOURCES_TYPE);
+
+    // check if the dropped element is valid
+    if (typeof type === 'undefined' || !type) {
+      return;
+    }
```

Comment:
Just check if `type` is an empty string or not for now.
According to the `getData` documentation "If there is no such data, returns the empty string"